contracts/access/AccessController.sol

```diff
@@ -174,26 +174,27 @@ contract AccessController is IAccessController, ProtocolPausableUpgradeable, UUP
         if (functionPermission == AccessPermission.ALLOW) {
             return;
         }
-
-        // If specific function permission is ABSTAIN, check module level permission
-        if (functionPermission == AccessPermission.ABSTAIN) {
-            uint8 modulePermission = getPermission(ipAccount, signer, to, bytes4(0));
-            // Return true if allow to call all functions of the module
-            if (modulePermission == AccessPermission.ALLOW) {
-                return;
-            }
-            // If module level permission is ABSTAIN, check transaction signer level permission
-            if (modulePermission == AccessPermission.ABSTAIN) {
-                // Pass if the ipAccount allow the signer can call all functions of all modules
-                // Otherwise, revert
-                if (getPermission(ipAccount, signer, address(0), bytes4(0)) == AccessPermission.ALLOW) {
-                    return;
-                }
-                revert Errors.AccessController__PermissionDenied(ipAccount, signer, to, func);
-            }
+        // Specific function permission is DENY, revert
+        if (functionPermission == AccessPermission.DENY) {
+            revert Errors.AccessController__PermissionDenied(ipAccount, signer, to, func);
+        }
+        // Function permission is UNSET, check module level permission
```

Comment:
I like the flatten in general.

There is one risk here.  The permission "ALLOW", and "DENY" were handled already.  You assumed functionPermission here is UNSET.  But if there is another permission introduced, this code may be a risk.

Reply:
You are correct, in `setPermission()` we have this limitation
```
        // permission must be one of ABSTAIN, ALLOW, DENY
        if (permission > 2) {
            revert Errors.AccessController__PermissionIsNotValid();
        }
```
So within the current code there are no other options.